spring-cloud-task-docs/src/main/asciidoc/features.adoc

```diff
@@ -314,6 +314,13 @@ https://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#boot-featu
 If so, it returns the exit code from  the `ExitCodeGenerator`. Otherwise, `1`
 is returned.
 
+In the case that an exception is thrown in an `onTaskStartup` method, the exit code for the application will be `1`.
+If an exception is thrown in either a `onTaskEnd` or `onTaskFailed`
+method, the exit code for the application will be the one established using the rules enumerated above.
+
+NOTE: In the case of an exception being thrown in a `onTaskStartup`, `onTaskEnd`, or `onTaskFailed`
+you can not override the exit code for the application using `ExitCodeExceptionMapper`.
+
```

Comment:
The `onTaskStartup`, `onTaskEnd`, etc are private methods and should be considered implementation details.  Therefore we should not be referring to them directly in our docs.

Reply:
Hmm, I'm not sure I understand, `onTaskStartup`  & `onTaskEnd` are not private as discussed here: https://docs.spring.io/spring-cloud-task/docs/current/reference/htmlsingle/#features-task-execution-listener

Reply:
My mistake.  I was thinking of the do* methods on the `TaskLifecycleListener`.